org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/syntaxcoloring/HighlightingStyles.xtend

```diff
@@ -21,5 +21,21 @@ interface HighlightingStyles {
 	val DEFAULT_ID = "default"
 	val INVALID_TOKEN_ID = "error"
 	val TASK_ID = "task"
+	
+	/**
+	 * A view of all default styles.
+	 *
+	 * @since 2.15
+	 */
+	val ALL_STYLES = #[
```

Comment:
In the org.eclipse.xtext.ui.editor.syntaxcoloring.IHighlightingConfiguration, we use an acceptor to collect all the known styles, e.g. the preference page in Eclipse is populated from the same source of information as the text attribute provider. Personally I prefer that compared to a static list of things that needs to be maintained. Why did you do it like that?

Reply:
> we use an acceptor to collect all the known styles
> ...
> Why did you do it like that?

I was not aware of it, thanks for the pointer.

Reply:
IHighlightedPositionAcceptor ???

Reply:
@cdietrich, are you suggesting something or what does this mean? If so, I did not get how does the `IHighlightedPositionAcceptor` solve the problem here. Could you please clarify this? Thanks!

Reply:
sry confused two problem. i am not sure if the 3 IHighlightingConfigurationAcceptor (2 existing + 1 new) could be unified somehow